src/black/linegen.py

```diff
@@ -1399,8 +1399,14 @@ def maybe_make_parens_invisible_in_atom(
     if is_lpar_token(first) and is_rpar_token(last):
         middle = node.children[1]
         # make parentheses invisible
-        first.value = ""
-        last.value = ""
+        if (
+            not node.children[1]
+            .prefix.strip()
+            .replace(" ", "")
+            .startswith("#type:ignore")
```

Comment:
Can you use `is_type_comment()` from `lines.py`?

Reply:
Umm I don't think it can be replaced directly with the `is_type_comment()` function, as it takes in a `leaf` and look at it's `value` field, while here the comment is as a `node`'s prefix (node.children[1] is still a node), I tried to find some existing function to convert this prefix string into a `leaf`, but I haven't found one yet... tbh I'm a bit confused here, any guidance would be really helpful.

Reply:
That's true, the interface doesn't apply. However, there is also a difference in behavior: the other function specifically looks for `# type: ignore`, but this one also allows `# type:ignore`. Mypy allows both so I think your solution is better, but we should be consistent (see also #2501 which asks us to format type-ignore comments).

I think we should:
- Add a new function, say `is_type_ignore_comment_string`, that takes a `str` and returns whether it's a type ignore comment. It should ignore whitespace.
- Split `is_type_comment()` into `is_type_comment()` and `is_type_ignore_comment()`
- Use `is_type_ignore_comment_string()` both here and from `is_type_ignore_comment()`.

Reply:
We also discussed this in #3339. I think for now let's not worry about `type:ignore` without the space and handle only `# type: ignore` with a space. We can figure out how to deal with the spacing issue in the other PR.

Reply:
Gotcha! Thanks for the guidance on design! I'll then keep it consistent with the rest part of Black to only handle the with-space case.

Just to clarify my understanding - when you suggest splitting `is_type_comment()` into `is_type_comment()` and `is_type_ignore_comment()`, does that mean we will be using the original `is_type_comment()` to handle non-ignore type of comments and use the new `is_type_ignore_comment()` to handle ignore-type of comments? (so all current usage of `is_type_comment()` for ignore-type comments will be using the new one instead)

Btw, I noticed that in #3339, it's been mentioned that the type comment is a deprecated feature, I tried to search for information on this deprecation but couldn't quite find where Python states about this deprecation. I'm wondering if you know if there's a particular PEP doc or other places that states it's a deprecated feature? I think it would also be beneficial to have this discussion documented here for future reference.

Reply:
Currently `is_type_comment()` takes a `suffix` argument, which is always passed as either `""` or `" ignore"`. Calls that use the former should stay as `is_type_coment()`, and calls that use the latter use `is_type_ignore_comment()`.

What's deprecated-ish is using type comments for type annotations like `x: []  # type: List[str]`. This was a compatibility feature introduced in PEP 484 for compatibility with Python 2 and Python 3.5, but it's not needed in modern versions of Python. Therefore, we're thinking of deprecating mypy support for it: python/mypy#12947. However, `# type: ignore` has no such replacement and it's here to stay.

Reply:
I have updated the code. I also removed the optional string parameter `suffix` in `is_type_coment()`, This parameter was only used to pass in `"ignore"`, and since the handling of ignore comments is now done by the new function `is_type_ignore_comment()`, there is no need to keep this parameter. Also, considering that `is_type_comment()` is responsible for handling only general types comments and there are plans to deprecate its support in the future, I added a note to its docstring stating that it might be deprecated in the future.